src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -15,125 +15,6651 @@
+    PHASE_THE_FORSAKEN_FRONT_II                         = 278, // Most likely after Rise Godfrey quest
+    PHASE_AMBER_MILL_I                                  = 279, // During Transdimensional Warfare: Chapter I and II quests
+    PHASE_AMBER_MILL_II                                 = 280, // After Transdimensional Warfare: Chapter III is completed but not rewarded
+    PHASE_AMBER_MILL_III                                = 281, // After Transdimensional Warfrase: Chapter III is rewarded
+    PHASE_THE_FORSAKEN_FRONT_III                        = 284, // During Cities in Dust quest
+    PHASE_THE_FORSAKEN_FRONT_IV                         = 289  // After Cities in Dust is completed or rewarded
+};
 
-enum eErland
+enum SilverpineTransports
 {
-    SAY_QUESTACCEPT     = 0,
-    SAY_START           = 1,
-    SAY_AGGRO           = 2,
-    SAY_PROGRESS        = 3,
-    SAY_LAST            = 4,
+    NPC_HORDE_ENGINEER                      = 44734,
+    NPC_SUBDUED_FOREST_ETTIN                = 44737,
+    NPC_FORSAKEN_TROOPER_F                  = 44732,
+    NPC_FORSAKEN_TROOPER_M                  = 44733,
+
+    SPELL_CHAIN_RIGHT                       = 83467,
+    SPELL_CHAIN_LEFT                        = 83464,
+    SPELL_EJECT_PASSENGERS_3_8              = 83477,
 
-    SAY_RANE            = 0,
-    SAY_RANE_ANSWER     = 5,
-    SAY_MOVE_QUINN      = 6,
+    EVENT_TO_FORSAKEN_HIGH_COMMAND          = 1,
+    EVENT_TO_THE_SEPULCHER                  = 2,
+    EVENT_TO_THE_FORSAKEN_FRONT             = 3,
+    EVENT_TO_DESPAWN                        = 4,
+    EVENT_YELL_1                            = 5,
+    EVENT_YELL_2                            = 6,
+    EVENT_YELL_3                            = 7,
 
-    SAY_QUINN           = 7,
-    SAY_QUINN_ANSWER    = 0,
-    SAY_BYE             = 8,
+    DATA_WAYPOINT_TO_FORSAKEN_HIGH          = 447310,
+    DATA_WAYPOINT_TO_SEPULCHER              = 447311,
+    DATA_WAYPOINT_TO_FORSAKEN_FRONT         = 447312,
+    DATA_WAYPOINT_TO_DESPAWN                = 447313,
 
-    QUEST_ESCORTING     = 435,
-    NPC_RANE            = 1950,
-    NPC_QUINN           = 1951
+    DATA_ANIMKIT_HORDE_ENGINEER             = 91298
 };
 
-class npc_deathstalker_erland : public CreatureScript
+// Horde Hauler - 44731
+struct npc_silverpine_horde_hauler : public ScriptedAI
 {
-public:
-    npc_deathstalker_erland() : CreatureScript("npc_deathstalker_erland") { }
+    npc_silverpine_horde_hauler(Creature* creature) : ScriptedAI(creature)
+    {
+        Initilize();
+    }
+
+    void Initilize()
+    {
+        if (Creature* engineer = me->FindNearestCreature(NPC_HORDE_ENGINEER, 15.0f))
+        {
+            if (Creature* ettin = me->FindNearestCreature(NPC_SUBDUED_FOREST_ETTIN, 15.0f))
+            {
+                engineer->CastSpell(ettin, SPELL_CHAIN_RIGHT, true);
+                engineer->CastSpell(ettin, SPELL_CHAIN_LEFT, true);
+                engineer->AddAura(DATA_ANIMKIT_HORDE_ENGINEER, engineer);
```

Comment:
DATA_ANIMKIT_HORDE_ENGINEER is a spell and the spell name is wrong. rename. And NO, AddAura is a no go. Also do NOT do stuff like this in Initialize and the constructor. Use JustAppeared() override for it.